lls/cache.c

```diff
@@ -333,7 +333,7 @@ icmp_cksum(void *buf, unsigned int size)
 }
 
 static int
-xmit_icmp_reply(struct gatekeeper_if *iface, struct rte_mbuf *pkt)
+xmit_icmp_ping_reply(struct gatekeeper_if *iface, struct rte_mbuf *pkt)
```

Comment:
I cannot comment inside of this function, but the block `if (rte_ipv4_frag_pkt_is_fragmented(icmp_ipv4)) { ... }` is no longer needed since `process_icmp_pkts()` takes care of this case.

Morever, hasn't the block `if (icmp_ipv4->next_proto_id != IPPROTO_ICMP) ...` also become unnecessary? If so, this function can now return `void`.

Reply:
Shouldn't the test `icmp_ipv4->next_proto_id != IPPROTO_ICMP` go to `process_icmp_pkts()`?

Reply:
Agreed that the two tests you mentioned in `xmit_icmp_ping_reply()` can be removed.

I don't think we need to test `icmp_ipv4->next_proto_id != IPPROTO_ICMP` in `process_icmp_pkts()`, because that condition will have already been checked either by the ACL rule or by the extension header matching function (`match_icmp()`).

Note that we still need the fragmentation check in `process_icmp_pkts()` because even though the extension header matching function (`match_icmp()`) checks for it, the ACL rule does not check for it.

Reply:
Your answer for both checks is a good fit to become comments in `process_icmp_pkts()` since they are not obvious.